lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -178,8 +183,8 @@ def refresh
     end
 
     def kill
-      if @session
-        run_sync("pkill -f #{remote_command_file('script')}")
+      if @pid_path
+        run_sync("pkill -P $(cat #{@pid_path})")
```

Comment:
`pkill` can read pids from a file on its own with `--pidfile`, there's really no need to fork a `cat`

Reply:
Ah, sorry, this was a bad suggestion. Reading the manpage I got the impression one could use both `-P` and `--pidfile` and make `pkill` kill processes which are children of processes listed in the file. Sadly it doesn't work like that :/